tests/utils.py

```diff
@@ -0,0 +1,207 @@
+from gptscript.command import exec
+from gptscript.tool import Tool
+import json
+from openai import OpenAI, Stream
+from openai.types.chat import ChatCompletion, ChatCompletionChunk
+import time
+import os
+
+
+def run_test(question, answer, dataset, judge_client):
+    tools = [
+        Tool(
+            tools=["retrieve"],
+            instructions="""
+
+                    {}
+                    """.format(question),
+        ),
+        Tool(
+            name="retrieve",
+            description="Retrieve information from a Knowledge Base Dataset",
+            args={"query": "Query to be executed against the Knowledge Base Dataset"},
+            instructions="""
+
+                #!knowledge retrieve -d {} -k 10 ${{query}}
+                """.format(dataset)),
+    ]
+
+    os.environ['GPTSCRIPT_INTERNAL_SYSTEM_PROMPT'] = ('You are an expert understanding context and extracting information')
+    resp = exec(tools)
+
+    if isinstance(resp, (list, tuple)):
+        returned_answer = ''.join(resp)
+    else:
+        returned_answer = resp
+    returned_answer = returned_answer.replace('\n', '')
+
+    match, reason = judge_final_answer(judge_client, False, returned_answer, answer)
+    assert match is True, """
+        Question: {},
+        Returned answer: {},
+        Expected answer: {},
+        Reason: {}
+        """.format(question, returned_answer, answer, reason)
+
+
+def judge_final_answer(judge_client: OpenAI, stream: bool, final_answer: str, final_answer_should: str) -> (bool, str):
```

Comment:
should I just export this from `function_benchmark` so we can import it here?

Reply:
Should be fine for now... i did tweak your test a little bit